### DISCUSSION TITLE
FTDI to flash ESP32 (UART1/UART2)

### DISCUSSION THREAD

**Discussion by pierregraciano** [micropython/micropython#15445] [ESP32]
Hello guys!

Id like to know if is possible to flash my ESP32, using a module FTDI (FT232RL) connected in UART1 or UART2 of ESP. I know that the USB connector is UART0. I tried to flash in UART0 using the module and occurred good. And Im using Thonny IDE to try this.

ESP model: ESP32 Doit Devkit - 38 pins

Thank you.

**Comment by davefes** [comment]
https://docs.micropython.org/en/latest/esp32/quickref.html

> The ESP32 has three hardware UARTs: UART0, UART1 and UART2. They each have default GPIO assigned to them, however depending on your ESP32 variant and board, these pins may conflict with embedded flash, onboard PSRAM or peripherals.
Any GPIO can be used for hardware UARTs using the GPIO matrix, except for input-only pins 34-39 that can be used as rx. To avoid conflicts simply provide tx and rx pins when constructing. The default pins listed below.

ESP32-WROOM-32U DevkitC-v1uses pins 34 and 35 for default UART0. For bare ESP32 modules I use those same pins.

Probably not much help, sorry.

**Reply by pierregraciano** [reply]
Hi Daves, I will see this doc, thank you so much.
I committed a type error, because I need to use UART1/2.

**Comment by glenn20** [comment]
My understanding (not verified) is that the ESP32 ROM bootloader accepts communication on UART0 or UART1, so it should be possible (eg. this [page](https://docs.espressif.com/projects/esptool/en/latest/esp32/advanced-topics/boot-mode-selection.html) implies it). Be careful that you board does not overload the UART1 gpio pins (9 and 10) for any other purpose (eg. for onboard QSPI flash).

**Reply by pierregraciano** [reply]
Hi Glenn. 
Thanks so much but I need another UART, sorry for the mistake. I need to use UART1/2. 
Using UART0 was good!

**Reply by glenn20** [reply]
I am saying you CAN use UART 1 (but not 2), unless your board uses the UART1 pins for some other reason.
